shapers/base.lua

```diff
@@ -67,8 +67,18 @@ function shaper:measureChar (char)
    local options = SILE.font.loadDefaults({})
    options.tracking = SILE.settings:get("shaper.tracking")
    local items = self:shapeToken(char, options)
-   if #items > 0 then
-      return { height = items[1].height, width = items[1].width, depth = items[1].depth }
+   SU.dump({ char, items })
+   if #items >= 1 then
+      local item = items[1]
+      if item.name == ".notdef" then
+         SU.warn(
+            ("No glyph available for '%s' in %s, returning measurements of '.notdef' instead"):format(
+               char,
+               options.family
+            )
+         )
+      end
+      return { height = item.height, width = item.width, depth = item.depth, name = item.name }
```

Comment:
Do we really need returning the name? Could be the GID, or just a boolean "undefined = true", so the caller wouldn't have to perform a ".notdef" explicit check. The name might not be meaningful or portable any for most extended characters (ligatures, etc.), would it?

Reply:
N.B. I am not fond of warnings is such utilities. The caller could try to measure char X, and fallback to char Y if X is not defined, without warnings.

Reply:
Yes I think we need to return the name. That way downstream users of the API can make their own call about whether to use the results or not. Actually I just took it a step farther and passed the entire table of info we got back from the shaper. That includes all the keys currently used and also some information about offsets that could be useful for some measurement scenarios. Also it includes the GID and name in case the consumer wants to check them.

My rational for including the warning was that we have better access to the font name and such here in the function than the user does easily when using the function so it was easier to create a meaningful warning. I'm not set on it though, so lets try a different shape ... returning a boolean status flag might be pretty useful too. How does that look?